simpleclient/src/main/java/io/prometheus/client/Summary.java

```diff
@@ -38,133 +40,231 @@
+        @Override
+        public Summary create() {
+            return new Summary(this);
+        }
+
+        public Builder quantiles(Quantile... quantiles) {
+            // We want the Summary to display quantiles in ascending order (e.g. 0.20, 0.50, 0.99)
+            //Arrays.sort(quantiles);
+            this.quantiles = quantiles;
+            this.shouldCalculateQuant = true;
+            return this;
+        }
+
+        public Builder stream(Stream stream) {
+            this.stream = stream;
+            this.shouldCalculateQuant = true;
+            return this;
+        }
     }
-  }
-
-  /**
-   *  Return a Builder to allow configuration of a new Summary.
-   */
-  public static Builder build() {
-    return new Builder();
-  }
-
-  @Override
-  protected Child newChild() {
-    return new Child();
-  }
-
-  /**
-   * Represents an event being timed.
-   */
-  public static class Timer {
-    Child child;
-    long start;
-    private Timer(Child child) {
-      this.child = child;
-      start = Child.timeProvider.nanoTime();
+
+    protected boolean shouldCalculateQuantiles() {
+        return shouldCalculateQuant;
     }
+
     /**
-     * Observe the amount of time in seconds since {@link Child#startTimer} was called.
-     * @return Measured duration in seconds since {@link Child#startTimer} was called.
+     * Return a Builder to allow configuration of a new Summary.
      */
-    public double observeDuration() {
-      double elapsed = (Child.timeProvider.nanoTime() - start) / NANOSECONDS_PER_SECOND;
-      child.observe(elapsed);
-      return elapsed;
-    }
-  }
-
-  /**
-   * The value of a single Summary.
-   * <p>
-   * <em>Warning:</em> References to a Child become invalid after using
-   * {@link SimpleCollector#remove} or {@link SimpleCollector#clear}.
-   */
-  public static class Child {
-    public static class Value {
-      private double count;  
-      private double sum;
+    public static Builder build() {
+        return new Builder();
     }
 
-    // Having these seperate leaves us open to races,
-    // however Prometheus as whole has other races
-    // that mean adding atomicity here wouldn't be useful.
-    // This should be reevaluated in the future.
-    private DoubleAdder count = new DoubleAdder();  
-    private DoubleAdder sum = new DoubleAdder();
 
-    static TimeProvider timeProvider = new TimeProvider();
+    @Override
+    protected Child newChild() {
+        if (this.shouldCalculateQuantiles()) {
+            if(this.stream == null){
+                this.stream = new CKMSStream(500, quantiles);
+            }
+            return new Child(this.stream);
+        } else {
+            return new Child();
+        }
+    }
+
     /**
-     * Observe the given amount.
+     * Represents an event being timed.
      */
-    public void observe(double amt) {
-      count.add(1);
-      sum.add(amt);
+    public static class Timer {
+        Child child;
```

Comment:
Should be marked *final*.
I haven't written this anywhere else, but you should mark variables final wherever possible:
fields, method vars etc.
There's an inspection for this in IntelliJ that can help you